### PR TITLE
fix(stream): fix TopN's usage of arguments

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
- remove `key_indices`, which is dead code for > 6 months
- add comments and fix usage for `order_pairs`. It already includes pk.
- add `order_by_len` to proto

### CODE REVIEW

File: src/stream/src/executor/group_top_n.rs [hunk 1]
```diff
@@ -130,11 +130,8 @@ pub struct InnerGroupTopNExecutorNew<S: StateStore, const WITH_TIES: bool> {
     /// group key -> cache for this group
     caches: HashMap<Vec<Datum>, TopNCache<WITH_TIES>>,
 
-    #[expect(dead_code)]
-    /// Indices of the columns on which key distribution depends.
-    key_indices: Vec<usize>,
-
-    sort_key_len: usize,
+    /// The length of the ORDER BY clause. Only used when `WITH_TIES` is true.
```
Comment (fuyufjh) [inline src/stream/src/executor/group_top_n.rs:133]:
```suggestion
    /// The number of fields of the ORDER BY clause. Only used when `WITH_TIES` is true.
```

File: src/stream/src/executor/group_top_n.rs [hunk 2]
```diff
@@ -130,11 +130,8 @@ pub struct InnerGroupTopNExecutorNew<S: StateStore, const WITH_TIES: bool> {
...
+    order_by_len: usize,
```
Comment (fuyufjh) [inline src/stream/src/executor/group_top_n.rs:134]: Could it be better to separate the _actual_ order-by columns and the PK columns? I had considered that before this PR

**Review by fuyufjh — APPROVED** [review]
Generally LGTM

File: src/stream/src/executor/group_top_n.rs (hunk 2 above)
Comment (xxchan) [reply src/stream/src/executor/group_top_n.rs:134]: https://github.com/risingwavelabs/risingwave/pull/4682 introduced `catalog.Table` in the proto. What's your idea? cc @shmiwy

**Pull request merged**